js/api/posts/create.mjs

```diff
@@ -0,0 +1,36 @@
+import { API_SOCIAL_URL } from "../constants.mjs";
+import { authFetch } from "../auth/authFetch.mjs";
+
+const action = "/posts";
+const method = "POST";  
+
+export async function createPost(postData) {
+    const createPostUrl = `${API_SOCIAL_URL}${action}`;
+
+    try {
+        const response = await authFetch(createPostUrl, {
+            method,
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(postData)
+        });
+
+        if (!response.ok) {
+            const error = await response.json();
+            throw new Error(`HTTP error! status: ${response.status}`);
+        }
+
+        const post = await response.json();
```

Comment:
There should be a conditional statement here to also check the `response` object for post.